blocks/popular-articles/popular-articles.js

```diff
@@ -1,3 +1,93 @@
-export default function decorate(block) {
-  block.innerHTML = '[Popular Articles Placeholder]';
+import { createOptimizedPicture, toClassName } from '../../scripts/lib-franklin.js';
+import { getCategory } from '../../scripts/scripts.js';
+
+async function fetchArticleData(paths) {
+  const PromiseArray = paths.map(async (path) => {
+    const res = await fetch(path);
+    const text = await res.text();
+
+    // Create a temporary element to extract the content within the <main> tag
+    const html = document.createElement('div');
+    html.innerHTML = text;
+
+    const catSlug = html.querySelector('meta[name="category"]').content;
+    const catData = await getCategory(toClassName(catSlug));
+
+    return {
+      image: html.querySelector('meta[property="og:image"]').content,
+      imageAlt: html.querySelector('meta[property="og:image:alt"]').content,
+      path,
+      title: html.querySelector('h1').textContent,
+      category: catData.Category,
+      categoryPath: catData.Path,
+    };
+  });
+
+  return Promise.all(PromiseArray);
+}
+
+async function getPathsFromSlideshow() {
+  const res = await fetch('/');
+  const html = await res.text();
+
+  // Create a temporary element to extract the content within the <main> tag
+  const tempElement = document.createElement('div');
+  tempElement.innerHTML = html;
+
+  // Get the content within the <main> tag
+  const slideShow = tempElement.querySelector('.slideshow');
+  const paths = [...slideShow.children].map((child) => new URL(child.querySelector('a').href).pathname);
+  paths.splice(3, paths.length);
+
+  return paths;
+}
+
+async function getPopularPosts(block) {
+  const res = await fetch('/popular-posts');
+  const text = await res.text();
+  const html = document.createElement('div');
+  let paths = [];
+  html.innerHTML = text;
+  // Get the content within the <main> tag
+  const heading = html.querySelector('h2');
+  block.innerHTML = heading.outerHTML;
+
+  const popularPostsElem = html.querySelector('.popularposts');
+
+  if (popularPostsElem) {
+    // eslint-disable-next-line max-len
+    paths = [...popularPostsElem.children].map((child) => new URL(child.textContent.trim()).pathname);
+  }
+  // if popularPostsElem is not found or there is less than three paths
+  if (paths.length !== 3) {
+    paths.push(...await getPathsFromSlideshow());
+    paths.splice(3, paths.length);
+  }
```

Comment:
I'd suggest adjusting to, so that if the author adds more than 3 rows, we truncate whatever overflows
```suggestion
  if (paths.length < 3) {
    paths.push(...await getPathsFromSlideshow());
  }
  paths.splice(3, paths.length);
```